### PR TITLE
seqexec config syntax

### PR DESCRIPTION
**Pull request by tpolecat**
This cleans up the syntax for extracting a value from a config, so instead of

```scala
extract[FPUnit](config, a)
```

You say

```scala
config.extract(a).as[FPUnit]
```

Debatable whether this is an actual improvement but I think it reads better.

### CODE REVIEW

**Comment by tpolecat** [comment]
BTW @arturog8m it's a little easier if you comment on the PR rather than on the commit. Not a big deal but it saves some clicking around.

**Comment by arturog8m** [comment]
Sounds good! :+1: 

Agreed on commenting on the PR instead of the commit - I meant to do that, but I didn't realized I was on the wrong place.

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TCS.scala [hunk 1]
```diff
@@ -137,13 +138,13 @@ object TCS {
 
   def fromSequenceConfig(config: Config)(s0: TcsConfig): TcsConfig = {
     List(
-      build(buildPwfs1Config, new ItemKey(TELESCOPE_KEY, GUIDE_WITH_PWFS1_PROP), config),
-      build(buildPwfs2Config, new ItemKey(TELESCOPE_KEY, GUIDE_WITH_PWFS2_PROP), config),
-      build(buildOiwfsConfig, new ItemKey(TELESCOPE_KEY, GUIDE_WITH_OIWFS_PROP), config),
-      build(buildOffsetConfig, new ItemKey(TELESCOPE_KEY, P_OFFSET_PROP), new ItemKey(TELESCOPE_KEY, Q_OFFSET_PROP),
-        config)
+      build(buildPwfs1Config,  TELESCOPE_KEY / GUIDE_WITH_PWFS1_PROP, config),
```
Comment (cquiroz) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TCS.scala]: Where are all the capital letter constants defined? If on the scala side we may rather use the capitalization of `TelescopeKey`, right?

Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TCS.scala]: Yes, I will look at caps in another PR.

**Comment by tpolecat** [comment]
Ok I added the property descriptor thing.

**Comment by cquiroz** [comment]
Looks good :+1:

**Pull request merged**